react/src/components/TableColumnsSettingModal.tsx

```diff
@@ -0,0 +1,135 @@
+import BAIModal, { BAIModalProps } from './BAIModal';
+import { SearchOutlined } from '@ant-design/icons';
+import { Checkbox, Input, theme, Form, Card } from 'antd';
+import { ColumnsType } from 'antd/es/table';
+import _ from 'lodash';
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+
+interface FormValues {
+  searchInput?: string;
+  selectedColumnKeys?: string[];
+}
+
+interface TableColumnsSettingProps extends BAIModalProps {
+  open: boolean;
+  onRequestClose: (formValues?: FormValues) => void;
+  columns: ColumnsType<any>;
+  selectKeys?: string[];
+}
+
+const TableColumnsSettingModal: React.FC<TableColumnsSettingProps> = ({
+  open,
+  onRequestClose,
+  columns,
+  selectKeys,
+  ...modalProps
+}) => {
+  const [form] = Form.useForm<FormValues>();
+  const { t } = useTranslation();
+  const { token } = theme.useToken();
+
+  const onChangeTitleToString: any = (element: any) => {
+    const text = React.Children.map(element.props.children, (child) => {
+      if (typeof child === 'string') {
+        return child;
+      }
+    });
+    return text;
+  };
+
+  const columnOptions = columns.map((column) => {
+    if (typeof column.title === 'string') {
+      return {
+        label: column.title,
+        value: _.toString(column.key),
+      };
+    } else if (typeof column.title === 'object' && 'props' in column.title!) {
+      return {
+        label: onChangeTitleToString(column.title),
+        value: _.toString(column.key),
+      };
+    } else {
+      return {
+        label: undefined,
+        value: _.toString(column.key),
+      };
+    }
+  });
+
+  return (
+    <BAIModal
```

Comment:
Please set the minimum height of the Modal to prevent the modal size from changing when the user types a filter.